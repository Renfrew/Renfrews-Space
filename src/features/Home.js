import React from 'react';

import Container from '@material-ui/core/Container';

const Home = (props) => {
  return (
    <Container component="main" disableGutters maxWidth={false}></Container>
  );
};

export default Home;
