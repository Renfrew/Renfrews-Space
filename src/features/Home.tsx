import React from 'react';
import { useHistory } from 'react-router-dom';
import { Theme, makeStyles, createStyles } from '@material-ui/core/styles';

import { Button, Container, Grid, Typography } from '@material-ui/core';

import Shenzhen from './asserts/shenzhen-city-night-unsplash.jpg';
import { Path } from './Constants';

const useStyles = makeStyles(({ breakpoints, spacing }: Theme) =>
  createStyles({
    container: {
      height: '100vh',
      display: 'flex',
      alignItems: 'center',
      backgroundImage: `url(${Shenzhen})`,
      backgroundPosition: 'center',
      backgroundRepeat: 'no-repeat',
      backgroundSize: 'cover',
    },
    parent: {
      marginBottom: spacing(20),
      [breakpoints.down('sm')]: {
        paddingLeft: '10%',
      },
      [breakpoints.up('sm')]: {
        paddingLeft: breakpoints.values.sm * 0.1 + 'px',
      },
    },
    text: {
      color: 'white',
      mixBlendMode: 'difference',
      filter: 'drop-shadow(.05em .05em #cf0c0c)',
    },
    explore: {
      textAlign: 'center',
      marginTop: spacing(6),
      [breakpoints.down('sm')]: {
        marginRight: '10%',
      },
      [breakpoints.up('sm')]: {
        marginRight: breakpoints.values.sm * 0.1 + 'px',
      },
    },
  })
);

const Home = () => {
  const classes = useStyles();
  const history = useHistory();

  function onExploreClick() {
    history.push(Path.AboutMe);
  }

  return (
    <Container
      component="main"
      disableGutters
      maxWidth={false}
      className={classes.container}
    >
      <Container
        component={Grid}
        container
        justifyContent="center"
        maxWidth="sm"
        className={classes.parent}
      >
        <Typography variant="h1" className={classes.text}>
          Renfrew
        </Typography>
        <Typography variant="h3" className={classes.text}>
          (Liang Chen)
        </Typography>
        <Typography variant="h6" className={classes.text}>
          I am a programer, a technology enthusiast.
        </Typography>
        <Grid className={classes.explore}>
          <Button
            onClick={onExploreClick}
            variant="contained"
            color="secondary"
            size="large"
          >
            Explore
          </Button>
        </Grid>
      </Container>
    </Container>
  );
};

export default Home;
