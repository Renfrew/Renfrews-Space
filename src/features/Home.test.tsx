import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('Testing the Home page', () => {
  it('The home page renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
