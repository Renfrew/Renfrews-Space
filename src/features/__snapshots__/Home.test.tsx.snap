// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home page The home page renders correctly 1`] = `
<main
  className="MuiContainer-root MuiContainer-disableGutters makeStyles-container-1 css-q7bdj4-MuiContainer-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiContainer-root MuiContainer-maxWidthSm makeStyles-parent-2 css-1vko90e-MuiGrid-root-MuiContainer-root"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-text-3 css-o2w69a-MuiTypography-root"
    >
      Renfrew
    </h1>
    <h3
      className="MuiTypography-root MuiTypography-h3 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-text-3 css-gepadz-MuiTypography-root"
    >
      (Liang Chen)
    </h3>
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInitial MuiTypography-alignInherit MuiTypography-displayInitial makeStyles-text-3 css-2ulfj5-MuiTypography-root"
    >
      I am a programer, a technology enthusiast.
    </h6>
    <div
      className="MuiGrid-root makeStyles-explore-4 css-u0fht3-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1olissi-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label css-8xplcm-MuiButton-label"
        >
          Explore
        </span>
      </button>
    </div>
  </div>
</main>
`;
